app/assets/javascripts/no_sunet.js

```diff
@@ -0,0 +1,43 @@
+$(document).on('click', '#no-sunet-button', function(){
+  setTimeout(function(){
+      $('#request_user_attributes_library_id').focus();
+  }, 1);
+
+  $('form').on('keypress', function(e) {
+    if(e.which == 13) {
+      $('.form-group input').checkForm(e);
+    }
+  });
+
+  $('.form-control').on('mouseenter mouseleave focus blur',function(e){
+    $('.form-group input').checkForm(e);
+  });
+
+  $('#no-sunet-submit').click(function(e) {
+    if ($('.form-group input').checkForm(e)){
+      $('form').submit();
+    }
+  });
+});
+
+$.fn.checkForm = function(e) {
+  var empty = true;
+
+  $(this).each(function() {
```

Comment:
Best I can tell, this simply loops through every element supplied that are not `#request_user_attributes_name` (in our context just `input`s) and checks to see if their value is present.  I'm not sure that is really what we ultimately want.

I think this would also cause an issue if a non-required input was in a `.form-group` element and not filled out, right?